src/liballoc/collections/btree/set.rs

```diff
@@ -1273,19 +1316,17 @@ impl<'a, T: Ord> Iterator for SymmetricDifference<'a, T> {
 
     fn next(&mut self) -> Option<&'a T> {
         loop {
-            match cmp_opt(self.a.peek(), self.b.peek(), Greater, Less) {
-                Less => return self.a.next(),
-                Equal => {
-                    self.a.next();
-                    self.b.next();
-                }
-                Greater => return self.b.next(),
+            match self.0.next() {
+                (None, None) => return None,
+                (Some(item), None) => return Some(item),
+                (None, Some(item)) => return Some(item),
+                (Some(_), Some(_)) => (),
             }
         }
     }
 
     fn size_hint(&self) -> (usize, Option<usize>) {
-        (0, Some(self.a.len() + self.b.len()))
+        (0, Some(self.0.a.len() + self.0.b.len()))
```

Comment:
Same as the union upper bound - this sum potentially overflows (and then it should be None).

Both can be ok but it would be best with a comment in that case, that max length of any set is limited.